### PR TITLE
Tech: Utilize Vanilla-Extract theme tokens

### PR DESCRIPTION
**Pull request by Youakeem**
## Describe your changes
Create and use a vanilla-extract CSS variables theme tokens.

## Justify why they are needed
Enable more style flexibility and easier implementation for things such as dark mode and contextual styling

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by Youakeem** [comment]
* **#4266** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4266?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Youakeem and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/app/[locale]/StoryblokLayout.css.ts [hunk 1]
```diff
@@ -1,16 +1,16 @@
 import { style } from '@vanilla-extract/css'
-import { colors } from 'ui/src/theme'
```
Comment (alebedev) [inline apps/store/src/app/[locale]/StoryblokLayout.css.ts:2]: Great if import from `ui` in css.ts did not break the build anymore, but I see CI errors.  Previous solution was importing from React-less `'ui/src/theme'`

**Review by alebedev — APPROVED** [review]
Looks good

**Review by guilhermespopolin — APPROVED** [review]
Looks good :)
Question: Wouldn't we need to use `createThemeContract` as mentioned in their [docs](https://vanilla-extract.style/documentation/theming/)? Assuming I want to update only two colors of the theme. Would something like this work or we'd need to "redefine" the whole theme? 🤔 

```
import { token } from 'theme.css'

export const additionalTheme = createTheme(tokens, {
  color: {
    primary: 'red',
    secondary: 'blue'
  }
})
```

**Comment by Youakeem** [comment]
> Looks good :)
> Question: Wouldn't we need to use `createThemeContract` as mentioned in their [docs](https://vanilla-extract.style/documentation/theming/)? Assuming I want to update only two colors of the theme. Would something like this work or we'd need to "redefine" the whole theme? 🤔 
> 
> ```
> import { token } from 'theme.css'
> 
> export const additionalTheme = createTheme(tokens, {
>   color: {
>     primary: 'red',
>     secondary: 'blue'
>   }
> })
> ```
> 
> 

Yes, we are gonna use `createThemeContract`. Will open a new PR soon.

**Comment by Youakeem** [comment]
### Merge activity

* **May 15, 10:11 AM EDT**: @Youakeem started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4266).

**Pull request merged**